cloudbuild-k8s-generator.py

```diff
@@ -0,0 +1,157 @@
+#!/usr/bin/env python
+#
+# Copyright 2018 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+
+CLOUDBUILD_CONFIG = 'cloudbuild.yaml'
+
+HEADER = """\
+##################################################################################
+## This file is generated by cloudbuild-k8s-generator.py. Do not manually edit. ##
+##################################################################################
+
+timeout: 1200s # 20m
+options:
+  machineType: 'N1_HIGHCPU_8'
+steps:
+"""
+
+INITIALIZE_GIT = """
+- id: Initialize Git
+  name: gcr.io/cloud-builders/git
+  entrypoint: bash
+  args:
+  - -exc
+  - |
+    # Cloud Build x GitHub integration uses source archives to fetch
+    # the source, rather than Git source fetching, and as a consequence
+    # does not include the .git/ directory. As a workaround, we clone
+    # the repository and reset it to this build's commit sha.
+    git clone https://github.com/GoogleCloudPlatform/click-to-deploy.git .tmp
+    mv .tmp/.git .git
+    rm -rf .tmp
+    git reset "$COMMIT_SHA"
+
+    # Download Git submodules
+    git submodule init
+    git submodule sync --recursive
+    git submodule update --recursive --init
+"""
+
+PULL_DEV_IMAGE = """
+- id: Pull Dev Image
+  name: gcr.io/cloud-builders/docker
+  waitFor:
+  - Initialize Git
+  dir: k8s/vendor/marketplace-tools
+  entrypoint: bash
+  args:
+  - -exc
+  - |
+    TAG="$$(./scripts/derive_tag.sh)"
+    docker pull "gcr.io/cloud-marketplace-tools/k8s/dev:$$TAG"
+    docker tag "gcr.io/cloud-marketplace-tools/k8s/dev:$$TAG" "gcr.io/cloud-marketplace-tools/k8s/dev:local"
+"""
+
+GET_KUBERNETES_CREDENTIALS = """
+# TODO(wgrzelak): Replace cluster-name and cluster-location with variables.
+- id: Get Kubernetes Credentials
+  name: gcr.io/cloud-builders/gcloud
+  args:
+  - container
+  - clusters
+  - get-credentials
+  - 'limani-integ'
+  - --region
+  - 'us-central1'
+  - --project
+  - '$PROJECT_ID'
+"""
+
+COPY_KUBECTL_CREDENTIALS = """
```

Comment:
Possibly delete these steps and use volumes per the discussions in our threads. This is good too.